src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -349,6 +350,20 @@ public void setEnableHeartbeat(boolean enableHeartbeat) {
     this.enableHeartbeat = enableHeartbeat;
   }
 
+  public void setHeartbeatFrequency(int frequency) {
```

Comment:
The setHeartbeatFrequency() function was added so it can get called by SessionUtil.updateSfDriverParamValues. Without creating and using this function, the heartbeat frequency was getting set as a connection parameter, but this parameter was not actually being used anywhere in the session.

This function updates the session variable heartbeatFrequency based on the connection parameter value given.